elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.pagination;
+
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulates the pagination strategy.
+ */
+
+@AllArgsConstructor
+@ToString
+public class Pagination {
+
+    /**
+     * Denotes the internal field names for paging.
+     */
+    public enum PaginationKey { page, pageSize }
+
+    public static final int DEFAULT_PAGE_SIZE = 500;
+    public static final int MAX_PAGE_SIZE = 10000;
```

Comment:
Since Elide is typically constructed in code, I would recommend a simple object that holds the parameters and is passed in to an Elide constructor. That said, @aklish may have some more in-depth thoughts on this.